contracts/rewards/StableTRIStaking.sol

```diff
@@ -0,0 +1,406 @@
+        require(_depositFeePercent <= 5e17, "StableTRIStaking: deposit fee can't be greater than 50%");
+        uint256 oldFee = depositFeePercent;
+        depositFeePercent = _depositFeePercent;
+        emit DepositFeeChanged(_depositFeePercent, oldFee);
+    }
+
+    /**
+     * @notice View function to see pending reward token on frontend
+     * @param _user The address of the user
+     * @param _token The address of the token
+     * @return `_user`'s pending reward token
+     */
+    function pendingReward(address _user, IERC20 _token) external view returns (uint256) {
+        require(isRewardToken[_token], "StableTRIStaking: wrong reward token");
+        UserInfo storage user = userInfo[_user];
+        uint256 _totalTRI = internalTRIBalance;
+        uint256 _accRewardTokenPerShare = accRewardPerShare[_token];
+
+        uint256 _currRewardBalance = _token.balanceOf(address(this));
+        uint256 _rewardBalance = _token == tri ? _currRewardBalance.sub(_totalTRI) : _currRewardBalance;
+
+        if (_rewardBalance != lastRewardBalance[_token] && _totalTRI != 0) {
+            uint256 _accruedReward = _rewardBalance.sub(lastRewardBalance[_token]);
+            _accRewardTokenPerShare = _accRewardTokenPerShare.add(
+                _accruedReward.mul(ACC_REWARD_PER_SHARE_PRECISION).div(_totalTRI)
+            );
+        }
+        return
+            user.amount.mul(_accRewardTokenPerShare).div(ACC_REWARD_PER_SHARE_PRECISION).sub(user.rewardDebt[_token]);
+    }
+
+    /**
+     * @notice Withdraw TRI and harvest the rewards
+     * @param _amount The amount of TRI to withdraw
+     */
+    function withdraw(uint256 _amount) external {
+        UserInfo storage user = userInfo[_msgSender()];
+        uint256 _previousAmount = user.amount;
+        require(_amount <= _previousAmount, "StableTRIStaking: withdraw amount exceeds balance");
+        uint256 _newAmount = user.amount.sub(_amount);
+        user.amount = _newAmount;
+
+        uint256 _len = rewardTokens.length;
+        if (_previousAmount != 0) {
+            for (uint256 i; i < _len; i++) {
+                IERC20 _token = rewardTokens[i];
+                updateReward(_token);
+
+                uint256 _pending = _previousAmount
+                    .mul(accRewardPerShare[_token])
+                    .div(ACC_REWARD_PER_SHARE_PRECISION)
+                    .sub(user.rewardDebt[_token]);
+                user.rewardDebt[_token] = _newAmount.mul(accRewardPerShare[_token]).div(ACC_REWARD_PER_SHARE_PRECISION);
+
+                if (_pending != 0) {
+                    safeTokenTransfer(_token, _msgSender(), _pending);
+                    emit ClaimReward(_msgSender(), address(_token), _pending);
+                }
+            }
+        }
+
+        internalTRIBalance = internalTRIBalance.sub(_amount);
+        tri.safeTransfer(_msgSender(), _amount);
+        _burn(_msgSender(), _amount);
```

Comment:
If a user has less than `amount` pTRI
He can never withdraw

If a user receives pTRI via a transfer then he will get zero TRI and reward tokens
Since user info is zero